src/x11/screen.c

```diff
@@ -82,27 +82,33 @@ void alloc_screen_ar(int n)
 
 int randr_event_base = 0;
 
+static int randr_major_version = 1;
+static int randr_minor_version = 5;
+
 void randr_init()
 {
         int randr_error_base = 0;
         if (!XRRQueryExtension(xctx.dpy, &randr_event_base, &randr_error_base)) {
                 fprintf(stderr, "Could not initialize the RandR extension, falling back to single monitor mode.\n");
                 return;
         }
+        XRRQueryVersion(xctx.dpy, &randr_major_version, &randr_minor_version);
         XRRSelectInput(xctx.dpy, RootWindow(xctx.dpy, DefaultScreen(xctx.dpy)), RRScreenChangeNotifyMask);
 }
 
 void randr_update()
 {
-        int n;
-        XRRMonitorInfo *m = XRRGetMonitors(xctx.dpy, RootWindow(xctx.dpy, DefaultScreen(xctx.dpy)), true, &n);
-
-        if (m == NULL || n == -1) {
```

Comment:
In [`xrandr.c`](https://cgit.freedesktop.org/xorg/app/xrandr/tree/xrandr.c#n2641) I see that there's an error check for `n == -1` so this check should probably stay and have a separate check for version.

Reply:
Hmm, looking at https://cgit.freedesktop.org/xorg/lib/libXrandr/tree/src/XrrMonitor.c it's checking a weird edge case where the server returns 0 monitors, but I guess that's possible. I'll put the check back.